### PR TITLE
Adds OSS::NumberInput along with tests & docs

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?
Adds OSS::NumberInput along with tests & docs

Related to: #https://github.com/upfluence/backlog/issues/2268

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/number-input.stories.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    onChange: {
+      description: 'A callback that sends back the new value of the input'
+    }
+  },
+  parameters: {
+    docs: {
+      description: {
+        component:
+          '[OPTIONAL] A number input wrapper with decrease & increase buttons. (Up & Right arrow increase, Down & Left arrow decrease)'
+      }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/number-input.stories.js:57]:
```suggestion
          'A number input wrapper with decrease & increase buttons. (Up & Right arrow increase, Down & Left arrow decrease)'
      }
```
Not necessary in components description

File: addon/components/o-s-s/number-input.stories.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      table: {
+        type: {
+          summary: 'number'
+        },
+        defaultValue: { summary: false }
+      },
+      control: { type: 'number' }
+    },
+    onChange: {
+      description: 'A callback that sends back the new value of the input'
```
Comment (JulienVannier66) [inline addon/components/o-s-s/number-input.stories.js:49]: `onChange` is optional in the ts, can you add `[OPTIONAL]` too

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

File: addon/components/o-s-s/number-input.hbs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="number-input fx-row">
+  <OSS::Button @square={{true}} @size="md" @icon="far fa-minus" {{on "click" this.decreaseValue}} />
+  <OSS::InputContainer @value={{this.localValue}} @onChange={{this.checkUserInput}} style={{this.dynamicWidth}}
+                       {{on "keydown" this.keyParser}} {{on "blur" this.checkUserInput}} />
```
Comment (phndiaye) [inline addon/components/o-s-s/number-input.hbs:4]: https://www.youtube.com/watch?v=SSbBvKaM6sk

**Review by phndiaye — APPROVED** [review]

**Pull request merged**